pkg/cmd/factory/factory_impl.go

```diff
@@ -0,0 +1,154 @@
+// Copyright 2021 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package factory
+
+import (
+	"crypto/tls"
+	"strings"
+	"time"
+
+	"github.com/pingcap/errors"
+	"github.com/pingcap/ticdc/cdc/kv"
+	cmdconetxt "github.com/pingcap/ticdc/pkg/cmd/context"
+	"github.com/pingcap/ticdc/pkg/security"
+	"github.com/pingcap/ticdc/pkg/version"
+	pd "github.com/tikv/pd/client"
+	"go.etcd.io/etcd/clientv3"
+	etcdlogutil "go.etcd.io/etcd/pkg/logutil"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/backoff"
+)
+
+type factoryImpl struct {
+	clientGetter ClientGetter
+}
+
+// NewFactory creates a client build factory.
+func NewFactory(clientGetter ClientGetter) Factory {
+	if clientGetter == nil {
+		panic("attempt to instantiate factory with nil clientGetter")
+	}
+	f := &factoryImpl{
+		clientGetter: clientGetter,
+	}
+
+	return f
+}
+
+func (f *factoryImpl) ToTLSConfig() (*tls.Config, error) {
+	return f.clientGetter.ToTLSConfig()
+}
+
+func (f *factoryImpl) ToGRPCDialOption() (grpc.DialOption, error) {
+	return f.clientGetter.ToGRPCDialOption()
+}
+
+func (f *factoryImpl) GetPdAddr() string {
+	return f.clientGetter.GetPdAddr()
+}
+
+func (f *factoryImpl) GetCredential() *security.Credential {
+	return f.clientGetter.GetCredential()
+}
+
+func (f *factoryImpl) EtcdClient() (*kv.CDCEtcdClient, error) {
+	ctx := cmdconetxt.GetDefaultContext()
+
+	tlsConfig, err := f.ToTLSConfig()
+	if err != nil {
+		return nil, err
+	}
+	grpcTLSOption, err := f.ToGRPCDialOption()
+	if err != nil {
+		return nil, err
+	}
+
+	logConfig := etcdlogutil.DefaultZapLoggerConfig
+	logConfig.Level = zap.NewAtomicLevelAt(zapcore.ErrorLevel)
```

Comment:
Should we consider to set log level based on parameters passed in cli?
A related scenario, PD is not accessible, user has to wait 30s (which is the default dial timeout we set below) and meets the error of `context.deadline exceeded`.

Reply:
Makes sense, we should standardize the log level of cli.